### ISSUE TITLE
[VandRS] Encoding/Decoding small object with Reed-Solomon Code

### ISSUE THREAD

**Issue by windkit** [leo-project/leo_jerasure#4]
Jerasure 2.0 require blocks aligned with respect to each other along a 16-byte boundary.

When we allocate memory for each data and code blocks, it only guaranteed to be aligned to page boundary and therefore could cause problem in GF-Complete operations.

As the problem only exists in small blocks (<512KB), and other solutions would probably need extra memory copy operations which could be bad for large object, I am currently forcing the minimum block size to 512KB.

In short,

Now, short objects suffer in performance degradation.

**Comment by windkit** [comment]
If mis-alignment is detected, it would change to pre-allocate one memory piece, reference the blocks to it. copy the input to the memory piece. After encoding, code blocks are copied back. As block size are multiples of 16-bytes, the problem would be solved.

**Comment by windkit** [comment]
TODO: Decoding
